### PR TITLE
🐛  Fix/news pages

### PR DESCRIPTION
**Pull request by RustySnek**

### CODE REVIEW

File: lib/kotkowo_web/live/news_live/all_news.ex [hunk 1]
```diff
@@ -13,10 +14,13 @@ defmodule KotkowoWeb.NewsLive.AllNews do
...
-      0 -> 1
-      _ -> max_page
-    end
+
+    max_page =
+      case max_page do
+        0 -> 1
+        _ -> max_page
+      end
+
```
Comment (ravensiris) [inline lib/kotkowo_web/live/news_live/all_news.ex]: Quick maths:

```elixir
max(1, max_page)
```

File: lib/kotkowo_web/live/news_live/latest_news.ex [hunk 2]
```diff
@@ -10,28 +10,34 @@ defmodule KotkowoWeb.NewsLive.LatestNews do
...
-      _ -> news
 
-    end
+    [header_news, tail] =
+      case length(news) do
+        0 -> [nil, []]
+        1 -> [hd(news), []]
+        _ -> news
+      end
+
```
Comment (ravensiris) [inline lib/kotkowo_web/live/news_live/latest_news.ex]: Terrible. You don't check length like in procedural languages in Elixir.

```elixir
{first_news, news} = List.pop_at(news, 0)
```

File: lib/kotkowo_web/live/news_live/latest_news.ex [hunk 3]
```diff
@@ -10,28 +10,34 @@ defmodule KotkowoWeb.NewsLive.LatestNews do
...
     popular_news = Enum.take(tail, 2)
```
Comment (ravensiris) [inline lib/kotkowo_web/live/news_live/latest_news.ex]: Use
```elixir
{popular_news, news} = Enum.split(news, 2)
```

File: lib/kotkowo_web/live/news_live/latest_news.ex [hunk 4]
```diff
@@ -10,28 +10,34 @@ defmodule KotkowoWeb.NewsLive.LatestNews do
...
 
     news =
       tail
       |> Enum.drop(2)
       |> Enum.take(4)
```
Comment (ravensiris) [inline lib/kotkowo_web/live/news_live/latest_news.ex]: Now just `Enum.take(news, 4)`

File: lib/kotkowo_web/live/news_live/latest_news.ex [hunk 5]
```diff
@@ -10,28 +10,34 @@ defmodule KotkowoWeb.NewsLive.LatestNews do
...
           |> Integer.to_string()
           |> StrapiClient.get_article_for_announcement()
+
         assign(socket, :article, article)
-    else
-      socket
-    end
+      else
+        socket
+      end
```
Comment (ravensiris) [inline lib/kotkowo_web/live/news_live/latest_news.ex]: Here's a cleaner solution
```elixir
defp get_article(%News{} = news) do
   case StrapiClient.get_article_for_announcement(news.id) do
      {:ok, %Article{} = article} -> article
      _ -> nil
   end
end

# ...
article = get_article(header)
socket = assign(socket, :article, article)
```

**Review by ravensiris — CHANGES_REQUESTED** [review]

**Review by RustySnek — COMMENTED** [review]
Done

**Review by ravensiris — APPROVED** [review]

**Pull request merged**